shell/platform/fuchsia/flutter/default_session_connection.h

```diff
@@ -74,15 +125,38 @@ class DefaultSessionConnection final : public flutter::SessionWrapper {
   // outstanding at any time. This is equivalent to how many times it has
   // called Present2() before receiving an OnFramePresented() event.
   const int kMaxFramesInFlight;
-  int frames_in_flight_ = 0;
+  fml::TimeDelta vsync_offset_;
 
+  int frames_in_flight_ = 0;
   int frames_in_flight_allowed_ = 0;
-
   bool present_session_pending_ = false;
 
-  void PresentSession();
+  // Flutter framework pipeline logic.
+
+  // The following fields can be accessed from both the raster and UI threads,
+  // so guard them with this mutex. If performance dictates, this could probably
+  // be made lock-free, but it's much easier to reason about with this mutex.
+  std::mutex mutex_;
+
+  // This is the last Vsync we submitted as the frame_target_time to
+  // FireCallback(). This value should be strictly increasing in order to
+  // guarantee that animation code that relies on target vsyncs works correctly,
+  // and that Flutter is not producing multiple frames in a small interval.
+  fml::TimePoint last_targeted_vsync_;
+
+  // This is true iff AwaitVSync() was called but we could not schedule a frame.
+  bool fire_callback_request_pending_ = false;
+
+  // The callback passed in from VsyncWaiter which eventually runs on the UI
+  // thread.
+  FireCallbackCallback fire_callback_;
+
+  // VsyncRecorder logic.
+
+  VsyncInfo GetCurrentVsyncInfo() const;
```

Comment:
Nit: Move this above all of the fields, with PresentSession and friends

Reply:
Done